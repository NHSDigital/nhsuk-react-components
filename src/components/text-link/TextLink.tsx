import React, { HTMLProps } from 'react';
import classNames from 'classnames';
import type { AsElementLink } from 'util/types/LinkTypes';

export type TextLinkProps = HTMLProps<HTMLAnchorElement>;

const TextLink: React.FC<AsElementLink<HTMLAnchorElement>> = ({ 
  children, 
  asElement: Component = 'a', 
  className, 
  ...rest 
}) => (
  <div className="nhsuk-action-link">
    <Component className={classNames('XXXnhsuk-action-link__link', className, 'text-link')} {...rest}>
      <span className="XXXnhsuk-action-link__text">{children}</span>
    </Component>
  </div>
);

export default TextLink;
