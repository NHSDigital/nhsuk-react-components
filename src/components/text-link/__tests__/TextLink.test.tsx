import React from 'react';
import { shallow } from 'enzyme';
import TextLink from '../TextLink';

describe('TextLink', () => {
  it('matches snapshot', () => {
    const element = shallow(<TextLink href="/test">Test</TextLink>);
    expect(element).toMatchSnapshot('TextLink');
    element.unmount();
  });
});
