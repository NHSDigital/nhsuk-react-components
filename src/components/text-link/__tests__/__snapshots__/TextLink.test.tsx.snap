// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink matches snapshot: TextLink 1`] = `
<div
  className="nhsuk-action-link"
>
  <a
    className="XXXnhsuk-action-link__link text-link"
    href="/test"
  >
    <span
      className="XXXnhsuk-action-link__text"
    >
      Test
    </span>
  </a>
</div>
`;
